### ISSUE TITLE
Running mtag example?

### ISSUE THREAD

**Issue by alanamarzoev** [p4lang/p4-spec#579]
Hi,

What is the suggested way to run the mtag example code? It appears that multiple switch definitions aren't supported by p4app, and the VM provided in the most recent tutorials only supports P4-16 programs. The only remaining option seems to be to install the relevant dependencies via the instructions in the SIGCOMM_2015 tutorial, and go from there. Am I missing something, or is this an accurate outlook? 

Thanks.

Edit: I see elsewhere that this isn't a full example.

**Comment by jnfoster** [comment]
TL;DR: yes, I would recommend installing bmv2 and p4c from source.

In more detail:

* I haven't confirmed this recently, but the mtag code should run out of
the box by using p4c-bm-ss (passing the --p4-14 flag) to produce a JSON
configuration for bmv2, and then running that either natively or via
Mininet.

* p4c and bmv2 can be installed on most Unix-like environments.

* I wouldn't go back to the SIGCOMM 2015 tutorial -- just follow the
instructions in p4lang/behavioral-model and p4lang/p4c.

* p4app does support multiple switches but yes, each switch currently needs
to run the same program.

-N

**Comment by jnfoster** [comment]
(Closing this issue now as it doesn't relate to the language specification.)

**Comment by alanamarzoev** [comment]
Thank you!
